### DISCUSSION TITLE
VI Mode? How to select and copy text?

### DISCUSSION THREAD

**Discussion by hxzhao527** [zellij-org/zellij#1315] [Q&A] [answered]
same as title.

Is there some way select and copy text without mouse?

**Comment by imsnif** [comment, marked as answer]
Not yet, unfortunately. Right now using the mouse is the only option. In our roadmap we're planning on giving you the option to temporarily open your default editor with the current pane's scrollbuffer in place. Then you can use it to copy, save the whole file elsewhere, search, whatever you wish.

**Comment by cosminadrianpopescu** [comment]
If you can set up your xclip to dump the content in a file, like this:

```bash
xclip -o >/tmp/clipboard
```

or, if for example, you are doing your copy from vim and you can dump your copied text into `/tmp/clipboard`, you have some options. Check out the `shell-expand-line` in `bash`. All this is for bash. I'm using vi mode and in my `.inputrc` I set up this: 

```bash
set keymap vi-insert 
"\C-r": shell-expand-line
```

Then, I can set up `zellij` to do the following:

```yaml
        - action: [WriteChars: $(cat /tmp/clipboard), Write: [18], SwitchToMode: Locked,]
          key: [Char: 'y']
```

As you can see, this will echo the `$(cat /tmp/clipboard)` subshell in the focused pane, then it will insert the escape code 18 (`C-r`, which is my shortcut for doing the subshell expansion) and it should paste at the cursor what you have in your `/tmp/clipboard` file. This should be the content of the clipboard.

If you use standard bash, the shortcut for  shell expansion is `ctrl+alt+e` and you will have to find out the escape code for this to replace the byte 18. Good luck. Let me know if you have questions.
